### DISCUSSION TITLE
Some Things I Don't Know How to do with Qtile Wayland

### DISCUSSION THREAD

**Discussion by MaybeAnonymous** [qtile/qtile#3400] [Q&A] [answered]
I use certain things on Xorg, and I am wondering how I'd use them on Wayland.
* DPI Scaling - how to change it, so the text and UI isn't so tiny **|** I'm using .Xresources on Xorg
* Keyboard Layout - how to set it to something else, like br-abnt2 **|** I'm using `setxkbmap -model abnt2 -layout br` on Xorg
* Alternatives to picom - for Vsync and Rounded Corners
* xss-lock alternatives - also an xsecurelock alternative, by the name it shouldn't work on Wayland properly 
* How to set an wallpaper - I'm using nitrogen right now, does feh work? **|** I'm using `nitrogen --set-auto "/path/to/wallpaper`
* How to separate config files for Xorg and Wayland **|** I have no idea how to do this
* Don't ask me why, but I'd like to remap mouse buttons | `xinput set-button-map "Compx 2.4G Receiver Mouse" 1 2 3 4 5 6 7 2 9` currently using this method

I have posted about this on Reddit, but someone recommended me to ask here, as I may get better support regarding Wayland

I will also look on the Sway documentation and wiki as there seems to be useful information there

**Comment by m-col** [comment, marked as answer]
Thanks for bringing these up. These are all great things that we could add to the docs, especially considering that many users (and devs) will use different things and so individually will miss some functionality (e.g. I don't use a screen locker... probably should).

Here are your answers:

- DPI: this is controlled via the output manager protocol, and so you need a utility to use as a client, for example `kanshi`, `wlr-randr or `wdisplays` (similar to `autorandr, `xrandr` and `arandr` respectively).
- Keyboard layout: As of `0.20.0` see [here](http://docs.qtile.org/en/v0.20.0/manual/wayland.html#keyboard-configuration) for docs. It can be set by XKBCOMMON environmental variables, or a `Core` command documented at the bottom of the page. (e.g. `qtile-cmd -o core -f set_keymap -a <args>`). From the next release, there will be better support for config-level keyboard configuration.
- vsync: Wayland is pixel perfect, so we get that for free on the rendering side (low-level within wlroots; higher-level within Qtile like synchronising clients is a WIP). Unless you mean _disabling_ it, in which case this is not yet a thing I don't think, but see: https://gitlab.freedesktop.org/wayland/wayland-protocols/-/merge_requests/65
- Screen locking: this is protocol based, so if we support the protocol, then you just need a client utility such as `swaylock`. See: https://github.com/qtile/qtile/pull/2852
- Wallpapers: Deep within the docs for the `Screen` config variable is information about setting wallpapers with Qtile directly: http://docs.qtile.org/en/v0.20.0/manual/config/screens.html#screen. Alternatively, you can use a layer-shell protocol client to set it, such as `swaybg`.
- Backend-specific configuration: See http://docs.qtile.org/en/v0.20.0/manual/wayland.html#backend-specific-configuration.
- Remapping mouse buttons: Good question! I suspect this is impossible currently, and I'm not sure what the best approach would be to supporting it, but it must be feasible. We can take a look at how `sway` does this for inspiration.

I'd like to emphasise that any feedback, especially on documentation, is very welcome.

**Reply by MaybeAnonymous** [reply]
Thanks! I like how Wayland is pixel perfect

**Reply by MaybeAnonymous** [reply]
Hey, I can't figure out DPI, I set the scale, but everything is blurry, is there an alternative for Xft.dpi in .Xresources?

**Reply by m-col** [reply]
No - I think this is the responsibility of the client, to support HiDPI. If the bar is blurry, then that of course _is_ our responsibility, but I don't know if anyone has looked at that yet.

**Reply by MaybeAnonymous** [reply]
The entire screen gets blurry, I'll search about it, but from what I've seen from a quick search it doesn't seem to be a Qtile problem

**Reply by clotodex** [reply]
Be aware of electron apps - most apps, even including chrome, need to be explicitly told that they are running on wayland and should behave like it @MaybeAnonymous
